test/distributed/org/apache/cassandra/distributed/impl/INodeProvisionStrategy.java

```diff
@@ -26,140 +26,69 @@
 
 import static org.apache.cassandra.utils.Shared.Recursive.INTERFACES;
 
+//TODO (now, correctness): one network changes the port where as multiple network changes the address... yet AbstractNodeProvisionStrategy always changes the port too!
 @Shared(inner = INTERFACES)
 public interface INodeProvisionStrategy
```

Comment:
I need to double check things, I do think I caused a regression here... I did enough to get my test to do what I needed... I will need to look closer here